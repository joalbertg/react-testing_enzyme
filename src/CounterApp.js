import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({value = 0}) => {
  const [counter, setCounter] = useState(value);
  const handleAdd = () => setCounter(counter => counter + 1);
  const handleSubtract = () => setCounter(counter - 1);
  const handleReset = () => setCounter(value);

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>
      <button onClick={ handleSubtract }>-1</button>
      <button onClick={ handleReset }>Reset</button>
      <button onClick={ handleAdd }>+1</button>
    </>
  );
}

CounterApp.propTypes = {
  value: PropTypes.number
}

export {
  CounterApp as default
}

