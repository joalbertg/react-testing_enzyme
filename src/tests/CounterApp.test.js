import React from 'react';
import { shallow } from 'enzyme';

import CounterApp from '../CounterApp';

describe('Test CounterApp component', () => {
  // se inicializa para no perder la referencia a los m√©todos
  // del tipo del wrapper dentro de las funciones
  // let wrapper; //undefined
  let wrapper = shallow(<CounterApp />);

  beforeEach(() => {
    wrapper = shallow(<CounterApp />);
  });

  test('Should display correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Should show default 100', () => {
    const value = 100;
    const wrapper = shallow(
      <CounterApp
        value={value}
      />
    );

    const counterText = wrapper.find('h2').text();

    expect(Number(counterText)).toBe(value);
  });

  test('Should increase +1', () => {
    //console.log(wrapper.find('button').at(2).html());
    wrapper.find('button').at(2).simulate('click');

    const counterText = wrapper.find('h2').text();

    expect(counterText).toBe('1');
  });

  test('Should decrease -1', () => {
    //console.log(wrapper.find('button').at(0).html());
    wrapper.find('button').at(0).simulate('click');

    const counterText = wrapper.find('h2').text();

    expect(counterText).toBe('-1');
  });

  test('Should reset value', () => {
    const value = 100;
    const wrapper = shallow(
      <CounterApp
        value={value}
      />
    );

    wrapper.find('button').at(2).simulate('click');
    wrapper.find('button').at(1).simulate('click');

    const counterText = wrapper.find('h2').text();

    expect(counterText).toBe('100');
  });
});

