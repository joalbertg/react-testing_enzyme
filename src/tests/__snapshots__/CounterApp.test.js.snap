// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CounterApp component Should display correctly 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    0
  </h2>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    +1
  </button>
</Fragment>
`;
